File: service-discovery-client/src/main/java/com/alekseysamoylov/servicediscoveryclient/controllers/ClientController.java
package com.alekseysamoylov.servicediscoveryclient.controllers;

import com.alekseysamoylov.servicediscoveryclient.config.ClientConfiguration;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class ClientController {

    private static final Logger LOGGER = LoggerFactory.getLogger(ClientController.class);

    @Value("${eureka.instance.metadataMap.region}")
    private String region;

    @Autowired
    private ClientConfiguration clientConfiguration;

    @GetMapping({"/ping", "/"})
    public String ping() {
        final String appName = clientConfiguration.getConfigurableName();
        LOGGER.info("Info access to client server " + appName + " with region " + region);
        LOGGER.debug("Debug access to client server " + appName + " with region " + region);
        return "The app: " + appName + " in region: " + region;
    }
}

File: kotlin-client/src/main/kotlin/com/alekseysamoylov/kotlinclient/controllers/ClientController.kt
package com.alekseysamoylov.kotlinclient.controllers

import com.alekseysamoylov.kotlinclient.config.ClientConfiguration
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.beans.factory.annotation.Value
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class ClientController {
    private val log = LoggerFactory.getLogger(ClientController::class.java)

    @Value("\${eureka.instance.metadataMap.region}")
    private lateinit var region: String

    @Autowired
    private lateinit var clientConfiguration: ClientConfiguration

    @GetMapping("/ping", "/")
    fun ping(): String {
        log.info("Info logging Kotlin client in region $region")
        log.debug("Debug logging Kotlin client in region $region")
        return "Hello kotlin app: ${clientConfiguration.getConfigurableName()} in region $region";
    }
}
